pkg/kubectl/cmd/testing/fake.go

```diff
@@ -675,6 +680,15 @@ func (f *fakeAPIFactory) KubernetesClientSet() (*kubernetes.Clientset, error) {
 	return clientset, f.tf.Err
 }
 
+func (f *fakeAPIFactory) MetricsClientSet() (metricsclientset.Interface, error) {
+	fakeClient := f.tf.Client.(*fake.RESTClient)
+	clientset := metricsclientset.NewForConfigOrDie(f.tf.ClientConfig)
+
+	clientset.MetricsV1alpha1().RESTClient().(*restclient.RESTClient).Client = fakeClient.Client
```

Comment:
we've got fake clientsets generated for the metrics clients.  Do you really need to do this?

Reply:
Tried that and getting this error:

```
error: no kind "NodeMetricsList" is registered for version "metrics/v1beta1"
```

Any idea?

Reply:
hmm... not off the top of my head.  I'll have to try and reproduce locally

Reply:
I'll prepare a branch to share to show you what I mean.

Reply:
Not sure what I did wrong there, but I managed to get it to work. Do you mind re-reviewing @DirectXMan12 ? :) (oh and happy new year! :tada:)